### PR TITLE
Add feature note about the ilab command redesign

### LINKED ISSUE #1333: Update Changelog to reflect ilab command redesign
following up to the command redesign, the release changelog should reflect the changes. I am opting to follow up in a separate PR as I don't want to add another commit to the already large litany of additions in #990

### PR DESCRIPTION
**Pull request by cdoern**
This adds a feature note about the redesign as requested by @leseb. Please let me know if this is not descriptive enough!

resolves #1333

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,4 +2,10 @@
 
 ### Features
 
+#### ilab command redesign
+
+The ilab command redesign included in v0.17 introduces a new command structure that follows a resource group design. This means that commands that once were something like `ilab chat` now are `ilab model chat`. The new groups are model, data, taxonomy, and config. The commands that fall under these are all of the pre-existing `ilab` commands just now grouped by the resource which the command commonly deals with.
+
+The old command structure is still aliased to work but will be removed in a future release.
```
Comment (leseb) [inline CHANGELOG.md:9]: I think it'd be good to give people a timeline for this. Let's coordinate during the community call.

**Review by leseb — CHANGES_REQUESTED** [review]
LGTM, small nit on the removal timeline.

File: CHANGELOG.md (hunk 1 above)
Comment (leseb) [reply CHANGELOG.md:9]: Slack thread if anyone wants to jump in: https://instruct-lab.slack.com/archives/C06TJP5V726/p1718180412862799

Comment (nathan-weinberg) [reply CHANGELOG.md:9]: @cdoern let's go ahead and target `0.19.0` for this?

**Comment by leseb** [comment]
I believe this is a must-have before releasing 0.17 (today?)

**Comment by nathan-weinberg** [comment]
@leseb we will probably do the release tomorrow - @cdoern can you implement this change so we can merge it before then?

**Comment by cdoern** [comment]
done!

**Review by leseb — APPROVED** [review]

**Pull request merged**